gamemodes/terrortown/gamemode/server/sv_player.lua

```diff
@@ -1430,14 +1430,21 @@ function GM:PlayerTakeDamage(ent, infl, att, amount, dmginfo)
 	-- start painting blood decals
 	util.StartBleeding(ent, dmginfo:GetDamage(), 5)
 
-	-- general actions for pvp damage
+	-- handle damage scaling by karma
 	if ent ~= att and IsValid(att) and att:IsPlayer() and GetRoundState() == ROUND_ACTIVE and math.floor(dmginfo:GetDamage()) > 0 then
-
 		-- scale everything to karma damage factor except the knife, because it
 		-- assumes a kill
 		if not dmginfo:IsDamageType(DMG_SLASH) then
 			dmginfo:ScaleDamage(att:GetDamageFactor())
 		end
+	end
+
+	-- before the karma is calculated, but after all other damage hooks / damage change is processed,
+	-- the armor system should come into place (GM functions are called last)
+	ARMOR:HandlePlayerTakeDamage(ent, infl, att, amount, dmginfo)
```

Comment:
This should not be excluded from the upper if, instead it would be good, to include it, since the checks are useful for the armor handling too and it would remove the duplicate if.